src/client/datascience/notebook/vscodeNotebookController.ts

```diff
@@ -385,8 +386,11 @@ export class VSCodeNotebookController implements Disposable {
             }
             return await kernel.executeCell(cell);
         } catch (ex) {
+            const errorHandler = this.serviceContainer.get<IDataScienceErrorHandler>(IDataScienceErrorHandler);
+
             // If there was a failure connecting or executing the kernel, stick it in this cell
-            return displayErrorsInCell(cell, execution, ex.toString());
+            displayErrorsInCell(cell, execution, await errorHandler.getErrorMessageForDisplayInCell(ex));
+            return NotebookCellExecutionState.Idle;
```

Comment:
Wouldn't this be an error? Idle seems wrong?

Reply:
This is the execution state, not about the state after the execution , i.e. not execution result.
Valid values from VS Code are running, not running or Pending.